main.go

```diff
@@ -208,6 +212,16 @@ func main() {
 		}
 	}
 
+	if startK8sWatcher != nil && *startK8sWatcher {
+		if apiShutdown, err := cmd.StartK8sWatcher(logger); err != nil {
+			logger.Error("Error starting k8s-watcher server", slog.String("err", err.Error()))
```

Comment:
sorry for confusion, I meant that here it should return but it does not. If you don't look the logs carefully you may not know that K8sWatcher did not even started. I'm not sure that's what is actually intended.